presto-spark-base/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeExecutionPlanRewrite.java

```diff
@@ -0,0 +1,120 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.spark;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.cost.PlanNodeStatsEstimate;
+import com.facebook.presto.cost.StatsAndCosts;
+import com.facebook.presto.metadata.Metadata;
+import com.facebook.presto.metadata.MetadataManager;
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.TableScanNode;
+import com.facebook.presto.sql.planner.Plan;
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.facebook.presto.sql.planner.SubPlan;
+import com.facebook.presto.sql.planner.TypeProvider;
+import com.facebook.presto.sql.planner.assertions.PlanAssert;
+import com.facebook.presto.sql.planner.assertions.PlanMatchPattern;
+import com.facebook.presto.sql.planner.plan.NativeExecutionNode;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+import static com.facebook.presto.SystemSessionProperties.NATIVE_EXECUTION_ENABLED;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static com.facebook.presto.sql.planner.assertions.PlanMatchPattern.node;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
+import static org.testng.AssertJUnit.assertTrue;
+
+public class TestPrestoSparkNativeExecutionPlanRewrite
+        extends AbstractTestQueryFramework
+{
+    private static final Metadata METADATA = MetadataManager.createTestMetadataManager();
+
+    private void assertPlanMatch(Session session, PlanNode actual, PlanMatchPattern expected)
+    {
+        PlanAssert.assertPlan(
+                session,
+                METADATA,
+                (node, sourceStats, lookup, session2, types) -> PlanNodeStatsEstimate.unknown(),
+                new Plan(actual, TypeProvider.empty(), StatsAndCosts.empty()),
+                expected);
+    }
+
+    private void assertPlanNotMatch(Session session, PlanNode actual, PlanMatchPattern expected)
+    {
+        PlanAssert.assertPlanDoesNotMatch(
+                session,
+                METADATA,
+                (node, sourceStats, lookup, session2, types) -> PlanNodeStatsEstimate.unknown(),
+                new Plan(actual, TypeProvider.empty(), StatsAndCosts.empty()),
+                expected);
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return PrestoSparkQueryRunner.createHivePrestoSparkQueryRunner();
+    }
+
+    @Test
+    public void testSingleStagePlanFragment()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(NATIVE_EXECUTION_ENABLED, "true")
+                .build();
+
+        SubPlan subPlan = subplan("SELECT orderkey, custkey, orderstatus, totalprice, orderdate, orderpriority, clerk, shippriority, comment " +
```

Comment:
Dumb question: (not related to this PR) I'm trying to understand the stuff between PlanNode and SubPlan. What is the purpose of subplan and why can't it just be another plan node with children?

Reply:
> Dumb question: (not related to this PR) I'm trying to understand the stuff between PlanNode and SubPlan. What is the purpose of subplan and why can't it just be another plan node with children?

The `PlanNode` is the logical layer plan which doesn't reflect how it will be executed on distributed manner. It'll be further broke up into several sub PlanNode trees wrapped by the`PlanFragment`s.  The `SubPlan` is a container of a list of `PlanFragment` which come from a same original PlanNode.

Reply:
nit
```java
        SubPlan subPlan = subplan(
                "SELECT orderkey, custkey, orderstatus, totalprice, orderdate, orderpriority, clerk, shippriority, comment FROM orders_bucketed",
                session);